docs: cumulative changes (#49009)
metabase/metabase: docs/questions/query-builder/expressions/cumulative.md

| July     | 5       | 2 + 5 = 7         |
| November | 4       | 2 + 5 + 4 = 11    |

> When you have only one breakout in the query, Metabase will sort the data by the breakout column in ascending order (using the logic that your database uses for the column's data type). You can add a **Sort** block by the breakout column to change how Metabase accumulates the metric.

Reviewer: ```suggestion
When you have only one breakout in the query, Metabase will sort the data by the breakout column in ascending order (using the logic that your database uses for the column's data type). To change how Metabase accumulates the metric, you can add a **Sort** block by the breakout column.
```

Might be worth a line to what we mean by "change" in this context.

## Cumulative metrics with multiple breakouts

> Because cumulative metrics calculate its summaries based on previous rows, Metabase needs to determine what those previous rows are. Metabase will decide how to compute and display the cumulative metric based on whether you're grouping by a datetime column, and on the order of other breakouts:

Reviewer: ```suggestion
Because cumulative metrics calculate their summaries based on previous rows, Metabase needs to determine what those previous rows are. Metabase will decide how to compute and display the cumulative metric first based on whether you're grouping by a datetime column, then by the order of other breakouts:
```

### Queries with a datetime dimension

> If you use datetime dimension in the **Group by** block, Metabase will always accumulate along the datetime dimension and break out by any other fields in the **Group by** block (in the order in which the other breakout fields are specified in the **Group by** block):

Reviewer: ```suggestion
If you use a datetime dimension in the **Group by** block, Metabase will accumulate along the datetime dimension, then break out by any other fields in the **Group by** block (in order):
```

![Cumulative count by category, source, and created at](../../images/cumulative-date-category.png)

> If there are multiple datetime dimensions, including multiple groupings by the same datetime column, Metabase will accumulate along the dimension with the lowest granularity, regardless of the order in which the breakouts are specified. For example, if you're grouping by "Created At: Month" and "Viewed At: Week", Metabase will accumulate along "Viewed At: Week".

Reviewer: ```suggestion
If there are multiple datetime dimensions (including multiple groupings by the same datetime column), Metabase will accumulate along the the more granular dimension, regardless of their order. For example, if you're grouping by "Created At: Month" and "Viewed At: Week", Metabase will accumulate along "Viewed At: Week".
```
"Lowest" granularity could translate to "coarsest".

![Cumulative count by two datetime fields](../../images/cumulative-multiple-datetimes.png)

> Because for queries with a datetime dimension, the accumulation will always be along the datetime dimension, any additional **Sort** blocks for non-datetime fields will only affect the order of breakouts in the results and won't affect how the results are computed.

Reviewer: ```suggestion
In queries with a datetime dimension, **Sort** blocks for non-datetime fields won't affect how Metabase computes the results. Sort blocks will only affect the order of breakouts in the results.
```

Just to chunk the info.

### Queries without datetime dimension

> If there is no datetime field in the **Group by** block, Metabase will accumulate along the _last_ dimension specified in the **Group By** block, and break out by the other fields (in the order in which they are specified).

Reviewer: ```suggestion
If there is no datetime field in the **Group by** block, Metabase will accumulate along the _last_ dimension specified in the **Group By** block, and break out by the other fields in their order, from left to right.
```

Is that right?

![Cumulative count by two categorical dimensions](../../images/cumulative-no-datetime.png)

> By default, Metabase will use ascending sort for the dimension used for accumulation. You can add a **Sort** block for the accumulation dimension to change the order (this will change both how the cumulative metric is computed and how results are presented).

Reviewer: ```suggestion
By default, Metabase will use ascending sort for the dimension used for accumulation. You can add a **Sort** block for the accumulation dimension to change the order, which will change both how Metabase computes the cumulative metric and, how it presents the results.
```

![Cumulative count by two categorical dimensions with order of accumulation changed](../../images/cumulative-no-datetime-order.png)

> **Sort** by any field other than the last (accumulation) field will only affect the order of breakouts in the results, and will not affect how the results are computed.

Reviewer: ```suggestion
Sorting by any field _other than_ the last (accumulation) field only affects the order of breakouts in the results. It won't change how Metabase computes the results.
```
